src/runtime/rpc/rpc_endpoint.cc

```diff
@@ -623,6 +623,8 @@ class RPCEndpoint::EventHandler : public dmlc::Stream {
 
 RPCCode RPCEndpoint::HandleUntilReturnEvent(bool client_mode, RPCSession::FEncodeReturn setreturn) {
   RPCCode code = RPCCode::kCallFunc;
+
+  ICHECK(channel_ != nullptr) << "channel is already closed";
```

Comment:
Since this would be the first error message seen by a user/developer who runs into it, we should include more context.  Also, since it could result from usage rather than being an internal error, we should use `CHECK` instead of `ICHECK`.

```c++
CHECK(channel_) << "Expected connection to server " << name_ << " to be active, but the connection was previously closed";
```